### PR TITLE
[RDY] fix: segfault when pasting in term with empty buffer

### LINKED ISSUE #13955: nvim segfaults in terminal_paste() if yank buffer is empty
If I try to paste while I'm in the terminal and the yank buffer/register is empty, then nvim will segfault in function terminal_paste() in file terminal.c. The reason for this seems to be that the pointer y_buffer is not checked before it is fed to strlen(). 

- `nvim --version`: v0.5.0-dev+1089-gdc3ca16a9
- Operating system/version: Debian 10 and Windows 10
- `$TERM`: screen.xterm-256color

### Steps to reproduce using `nvim -u NORC`
1. Make sure your yank buffer/register is empty
2. Open terminal using `:term`
3. Press p to paste

### PR DESCRIPTION
**Pull request by dylanarmstrong**
Close #13955 by adding a check that buffer array is not empty before sending to terminal_paste.

### CODE REVIEW

File: test/functional/terminal/buffer_spec.lua [hunk 1]
```diff
@@ -258,6 +258,13 @@ describe(':terminal buffer', function()
   it('handles wqall', function()
     eq('Vim(wqall):E948: Job still running', exc_exec('wqall'))
   end)
+
+  it('does not segfault when pasting empty buffer #13955', function()
+    feed_command('terminal')
+    feed('<c-\\><c-n>')
+    feed_command('put a') -- buffer a is empty
+    eq(2, eval('1+1')) -- check nvim still running
```
Comment (vigoux) [inline test/functional/terminal/buffer_spec.lua:266]: I think we have something like `helper.alive()` to exactly check this kind of things.

**Review by vigoux — COMMENTED** [review]
Juste left a little comment, can't tell anything about the code though, but it LGTM.

File: test/functional/terminal/buffer_spec.lua (hunk 1 above)
Comment (dylanarmstrong) [reply test/functional/terminal/buffer_spec.lua]: Updated here and the other 2 tests as well in this file.

**Comment by dylanarmstrong** [comment]
Hey @vigoux / @bfredl is there anything else I need to do for this?
Thanks!

File: src/nvim/ops.c [hunk 2]
```diff
@@ -2971,7 +2971,7 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
     y_array = reg->y_array;
   }
 
-  if (curbuf->terminal) {
+  if (y_size > 0 && curbuf->terminal) {
```
Comment (mhinz) [inline src/nvim/ops.c:2974]: Move the check for `y_size` to the start of `terminal_paste()`.

Now we would run all the code below although we're in a terminal.

Comment (dylanarmstrong) [reply src/nvim/ops.c]: Done

**Comment by bfredl** [comment]
thanks :)

**Pull request merged**